test/integration/mysql_integration_test.rb

```diff
@@ -106,7 +106,8 @@ class AfterPersistedTest < MySQLIntegrationTest
     end
 
     test "can't be used to inject SQL using .where" do
-      assert_raises MySQLBinUUID::InvalidUUID do
+      expected_error = ActiveRecord.version.to_s < "7.1" ? MySQLBinUUID::InvalidUUID : ActiveRecord::StatementInvalid
+      assert_raises(expected_error) do
         MyUuidModel.where(the_uuid: "' OR ''='").first
       end
     end
```

Comment:
In Rails 7.1, the `MySQLBinUUID::InvalidUUID` error raised by this library gets wrapped in an `ActiveRecord::StatementInvalid`.